src/app/admin/services/controller/equipment-controller.service.ts

```diff
@@ -218,4 +192,68 @@ export class EquipmentController {
       this.statusIdsDictionary[status.name as keyof EquipmentStatusId] = status.id;
     });
   }
+
+  private manageEquipment(equipment?: Equipment) {
+    this.openEquipmentModal(equipment)
+      .pipe(
+        filter(Boolean),
+        switchMap((res) => {
+          return forkJoin({
+            photo: res.file ? this.api.uploadPhoto(res.file) : of(null),
+            equipment: of(res.equipment),
+          });
+        }),
+        switchMap((res) => {
+          if (res.photo) res.equipment.photoID = res.photo.data.id;
+          return equipment
+            ? this.api.editEquipment(res.equipment, equipment.id)
+            : this.api.registerEquipment(res.equipment);
+        }),
+        switchMap(() => this.fetchEquipments()),
+        untilDestroyed(this),
+      )
+      .subscribe((res) => {
+        this.notificationService.openSuccess(equipment ? 'Оборудование отредактированно' : 'Оборудование добавлено');
```

Comment:
Better to have notification messages as consts or enum

Reply:
fixed